lib/graph_records.rb

```diff
@@ -27,41 +210,71 @@ def graph(**objects)
     @inspected = Set.new
     @focus = @focus_config.map { _1.to_s.classify.constantize.where(id: _2) }
 
-    objects.map do |klass, ids|
-      class_name = klass.to_s.singularize
-      associated_objects =
-        load_association(
-          class_name,
-          class_name.classify.constantize.where(id: ids)
+    if @primary_type.nil?
+      @primary_type =
+        (
+          if objects.keys.size >= 1
+            objects.keys.first.to_s.singularize.to_sym
+          else
+            :patient
+          end
         )
+    end
+
+    begin
+      objects.map do |klass, ids|
+        class_name = klass.to_s.singularize
+        associated_objects =
+          load_association(class_name.classify.constantize.where(id: ids))
 
-      @focus += associated_objects
+        @focus += associated_objects
 
-      associated_objects.each do |obj|
-        @nodes << obj
-        inspect(obj)
+        associated_objects.each do |obj|
+          @nodes << obj
+          introspect(obj)
+        end
+      end
+      ["flowchart TB"] + render_styles + render_nodes + render_edges +
+        (@clickable ? render_clicks : [])
+    rescue StandardError => e
+      if e.message.include?("Recursion limit")
+        # Create a Mermaid diagram with a red box containing the error message.
+        [
+          "flowchart TB",
+          "    error[#{e.message}]",
+          "    style error fill:#f88,stroke:#f00,stroke-width:2px"
+        ]
+      else
+        raise e
       end
     end
-    ["flowchart TB"] + render_styles + render_nodes + render_edges
   end
 
   def traversals
-    @traversals ||= {
-      patient: %i[parents consents class_imports cohort_imports],
-      parent: %i[consents class_imports cohort_imports]
-    }.merge(@traversals_config)
+    @traversals ||=
+      (DEFAULT_TRAVERSALS[@primary_type] || {}).merge(@traversals_config)
   end
 
   def render_styles
-    {
-      patient: "fill:#c2e598",
-      parent: "fill:#faa0a0",
-      consent: "fill:#fffaa0",
-      class_import: "fill:#7fd7df",
-      cohort_import: "fill:#a2d2ff",
-      patient_focused: "fill:#c2e598,stroke:#000,stroke-width:3px",
-      parent_focused: "fill:#faa0a0,stroke:#000,stroke-width:3px"
-    }.with_indifferent_access
+    object_types =
+      @nodes.to_a.map { |node| node.class.name.underscore.to_sym }.uniq
```

Comment:
You can `map` on a `Set`.

```suggestion
      @nodes.map { |node| node.class.name.underscore.to_sym }.uniq
```